src/Raven.Server/ServerWide/Commands/ConnectionStrings/PutConnectionStringCommand.cs

```diff
@@ -70,6 +70,8 @@ public PutSqlConnectionStringCommand(SqlConnectionString connectionString, strin
 
         public override void UpdateDatabaseRecord(DatabaseRecord record, long etag)
         {
+            // Validate ConnectionString.FactoryName
+            SqlProviderParser.GetSupportedProvider(ConnectionString.FactoryName);
```

Comment:
I think this should be done in the endpoint, not in the command

Reply:
I think it should be added here:

https://github.com/ravendb/ravendb/blob/08a02b747b8096510800b21979fd1ca2419a7257/src/Raven.Client/Documents/Operations/ETL/SQL/SqlConnectionString.cs#L15-L19

Reply:
@arekpalinski I've tried to put it here, but it won't throw exception on put. It's used to validate it later, e.g. when adding an ETL task

Reply:
> I think this should be done in the endpoint, not in the command

That was also my first thought, to put the validation inside the [switch from the endpoint](https://github.com/ravendb/ravendb/blob/f8ff7931a26f923c207154c0837824014af15047/src/Raven.Server/Web/System/OngoingTasksHandler.cs#L780-L786), by adding another case for Sql strings. However we don't have a `SqlConnectionString` instance yet there, so I'd need to deserialize the `Blittable` first. It's deserialized [while](https://github.com/ravendb/ravendb/blob/f49607280d50bda295e368fc44f4d77e78ae5b18/src/Raven.Server/ServerWide/ServerStore.cs#L2219-L2221) creating `PutSqlConnectionStringCommand` instance. I've looked for an opportunity to avoid doing it unnecessarily one more time by myself. 

I've put validation inside the command, where I can look at the `FactoryName` property already. I think that's the cleanest way to go, but if we want to do it faster, I'll read `FactoryName` from the `Blittable` inside endpoint.